regnety/models/model.py

```diff
@@ -0,0 +1,69 @@
+import tensorflow as tf
+
+from regnety.regnety.config import get_model_config, ALLOWED_FLOPS
+from regnety.regnety.models.blocks import PreStem, Stem, Stage, Head
+
+class RegNetY(tf.keras.Model):
+    """
+    RegNetY model class. Subclassed from tf.keras.Model. Instantiates a randomly
+    initialised model (to be changed after training). User must provide `flops`
+    argument to model.
+
+    Args:
+        flops: Flops of the model eg. "400MF"
+    """
+
+    def __init__(self, flops:str = ""):
+        super(RegNetY, self).__init__()
+
+        if flops not in ALLOWED_FLOPS:
+            raise ValueError("`flops` must be one of " + str(ALLOWED_FLOPS))
+        
+        self.flops = flops
+        self.config = get_model_config(self.flops)
+        self.model = self._get_model_with_config(self.config)
+
+    def build(self, input_shape):
+        self.model.build(input_shape)
+
+    def call(self, inputs):
+        self.model.call(inputs)
+    
+    def get_model(self):
+        """Returns sequential model constructed in this class"""
+        return self.model
+    
+
+    def _get_model_with_config(self, config):
+        """
+        Makes a tf.keras.Sequential model using the given config.
+         
+        """
+        model = tf.keras.models.Sequential()
+        model.add(tf.keras.layers.InputLayer(input_shape = (224,224,3)))
```

Comment:
Should we get this shape from a configuration file? I'd also suggest looking for the `keras.applications` classes since they allow variable-shape images during inference. If we prefix this here then that possibility gets nullified. I think allowing a model to do inference on shapes not discovered during training is quite practical.

WDYT?

Reply:
Sure. I'll add an `input_shape` arg in \_\_init\_\_ and make sure it is >= 224. If not specified, will default to 224. I believe this is the gist of what they've done in keras.applications.

Reply:
Yes, sounds good to me.